ballerina/tests/from_json_string_test.bal

```diff
@@ -60,6 +60,17 @@ isolated function testSimpleJsonStringToRecord() returns Error? {
     test:assertEquals(recC.get("b"), 1);
 }
 
+type ReadOnlyUser readonly & record {|
+    int id;
+|};
+
+@test:Config
+isolated function testSimpleJsonStringToRecord2() returns Error? {
+    string user = string `{"id": 4012}`;
+    ReadOnlyUser r = check parseString(user);
+    test:assertEquals(r, {id: 4012});
+}
+
```

Comment:
Can we add more tests with
- members of selectively-immutable types that become immutable because the type it is in is immutable - e.g., like the `id` field, but with a selectively-immutable type
- field of a readonly intersection type in a mutable type

Reply:
Added tests for these scenarios.

Reply:
Can you please point to the relevant tests for each of these cases?

Reply:
`testSimpleJsonToRecord6 , testSimpleJsonStringToRecord6` test functions
`testSimpleJsonToRecord3, testSimpleJsonStringToRecord3 ` test functions

Reply:
> * field of a readonly intersection type in a mutable type

Still can't find tests like this?

Reply:
```
type ReadOnlyUsersRecord3 readonly & record {|
    ReadonlyUserId[] ids;
    ReadonlyUserName[] names;
|};

@test:Config
isolated function testSimpleJsonToRecord6() returns Error? {
    json user = {ids: [4012, 4013], names: [{firstname: "John", lastname: "Doe"},
                                            {firstname: "Jane", lastname: "Doe"}], age: 27};
    ReadOnlyUsersRecord3 r = check parseAsType(user);
    test:assertEquals(r, {ids: [4012, 4013], names: [{firstname: "John", lastname: "Doe"},
                                                    {firstname: "Jane", lastname: "Doe"}]});
}
```
In here ReadonlyUserName is a readobly type of a record type

Reply:
What I meant was something like

```
type T1 record {
   T2[] & readonly v;
};
```

Reply:
Added tests, `testSimpleJsonToRecord7`, testSimpleJsonStringToRecord7

Reply:
The type is still readonly in it?

Reply:
Updated.
https://github.com/ballerina-platform/module-ballerina-data.jsondata/pull/45/commits/6ca0ca5b8fa44cb89b3155a68d59ef9d9bebc841

```
type ReadOnlyUsersRecord5 record {|
    ReadonlyUserId[] & readonly ids;
    ReadonlyUserName[] & readonly names;
|};
```